docarray/array/mixins/evaluation.py

```diff
@@ -65,19 +67,17 @@ def evaluate(
                     f'Document {d!r} or {gd!r} has no matches, please check your Document'
                 )
 
-            desired = {hash_fn(m) for m in gd.matches}
-            if len(desired) != len(gd.matches):
+            targets = gd.matches[:max_rel]
+            desired = {hash_fn(m) for m in targets}
+            if len(desired) != len(targets):
                 warnings.warn(
                     f'{hash_fn!r} may not be valid, as it maps multiple Documents into the same hash. '
```

Comment:
I'm not sure what is the desired behaviour here but it looks like the warning is no more reachable.